rasa/cli/train.py

```diff
@@ -144,3 +159,58 @@ def extract_additional_arguments(args: argparse.Namespace) -> Dict:
         arguments["debug_plots"] = args.debug_plots
 
     return arguments
+
+
+def _enrich_config(
+    config_path: Text, missing_keys: List[Text], FALLBACK_CONFIG_PATH: Text
+):
+    import rasa.utils.io
+
+    config_data = rasa.utils.io.read_yaml_file(config_path)
+    fallback_config_data = rasa.utils.io.read_yaml_file(FALLBACK_CONFIG_PATH)
+
+    for k in missing_keys:
+        config_data[k] = fallback_config_data[k]
+
+    rasa.utils.io.write_yaml_file(config_data, config_path)
+
+
+def _get_valid_config(
+    config: Optional[Text],
+    mandatory_keys: List[Text],
+    default_config: Text = DEFAULT_CONFIG_PATH,
+) -> Text:
+    if config:
+        # config is provided via '-c' argument. config file needs to contain
+        # all mandatory keys
+        if not os.path.exists(config):
+            print_error(
+                "The config file '{}' does not exist. Use '--config' to specify a "
+                "valid config file."
+                "".format(config)
+            )
+            exit(1)
+
+        missing_keys = missing_config_keys(config, mandatory_keys)
+        if missing_keys:
+            print_error(
+                "Configuration file '{}' is missing mandatory parameters: "
+                "'{}'. Add missing parameters to configuration file and try again."
+                "".format(config, "', '".join(missing_keys))
+            )
+            exit(1)
+        return config
+
+    # use default config path and enrich it in case mandatory keys are missing
+    config = get_validated_path(default_config, "config", FALLBACK_CONFIG_PATH)
+    missing_keys = missing_config_keys(config, mandatory_keys)
+
+    if missing_keys:
+        print_warning(
+            "Configuration file '{}' is missing mandatory parameters: "
+            "'{}'. Filling missing parameters from fallback configuration file: '{}'."
+            "".format(config, "', '".join(missing_keys), FALLBACK_CONFIG_PATH)
+        )
+        _enrich_config(config, missing_keys, FALLBACK_CONFIG_PATH)
```

Comment:
is there a reason why we only do that if we load from the default path but not from a user specified path?

Reply:
@vova please correct me if I'm wrong.
If you specify a path, you normally touched the default config in some sense. If the provided config is invalid, you want to be informed about that. You do not want to start training with some default values. 
However, if you do not specify a config path, you just want to use the default config. If the default config is invalid, you nevertheless want to start the training process. So, we enrich the config.

Reply:
mhm but by not specifying a path, you are not using the *default config* but rather the *default config path* - it might still be your custom config, just at the default location.

Reply:
I had similar concerns when starting to implement this, but I did not push further. Maybe @Ghostvv can explain better.

Reply:
I'm a bit confused now 🙈. So if `-c` is not provided, but there is custom config, but it is not full, we enrich it or raise? this is the question?

Reply:
Per current implementation we enrich it. However, if you provide `-c config.yml` we would raise.
Basically, the question is, what was the intention behind your request?

Reply:
my intention was if user uses custom configuration and it is wrong, we should raise, because most probably it was unintentional mistake, and we want to catch it as early as possible

Reply:
How should we proceed?
I guess we have the following options:
1. Merge as it. Only configs that are defined via `-c` are defined as "custom" and errors might be raised. If default config path is used, no errors will be raised, just some warnings get printed.
2. Work on a way to detect a "custom" config (e.g. looks different than the config generated by `rasa init`). Raise error if "custom" config is invalid.
3. Keep as it is: Print warnings and enrich config if needed.

I would prefer option 2 or 3.

Reply:
mhm so the default config should have all values, so it should never raise a warning right? so only if you change it and remove items, you'd run into the issue of missing required keys, right? so why not always raise an error if the config isn't complete? on the other hand a user might want to remove a key to always use the default 🤔

Reply:
@ella initially suggested the idea of enriching the config as user where missing keys in their config due the merge. before you just needed, for example, the policy key if you wanted to train a core model. now, if you train a rasa model this would fail as the mandatory keys for nlu are missing. thus, the idea was to print a warning an enrich the config.

Reply:
If our user are now educated enough (in a sense that they now know what keys a config should contain), I'm also fine with always raising an error.

Reply:
I think we should educate them, rather than doing this job for them, where at later stage they might experience unexpected behavior due to this rather hidden process

Reply:
I agree, I am also in favor of failing early - this is a breaking change though

Reply:
Ok, so I remove the functionality of enriching the config and will always throw an error if the provided config is invalid. I'm going to update the PR tomorrow.